### PR TITLE
added splice-call feature in goto instrument

### PR DESCRIPTION
**Pull request by KPouliasis**
it prepends a nullary function call say foo in the body of another call say moo when called as --splice-call moo,foo  . Also added tests

### CODE REVIEW

File: regression/splice_call_01/main.c [hunk 1]
```diff
@@ -0,0 +1,9 @@
+void moo()
+{
+    return;
+}
```
Comment (kroening) [inline regression/splice_call_01/main.c]: stupid nit: two extra spaces

File: src/goto-instrument/goto_instrument_parse_options.cpp [hunk 2]
```diff
@@ -1448,6 +1449,19 @@ void goto_instrument_parse_optionst::instrument_goto_program()
       full_slicer(goto_functions, ns);
   }
 
+  // splice  option
+  if(cmdline.isset("splice-call"))
```
Comment (kroening) [inline src/goto-instrument/goto_instrument_parse_options.cpp]: extra space between splice and option

File: src/goto-instrument/splice_call.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  split_string(callercallee, ',', result);
+  return (result.size()!= 2);
+}
+
+bool splice_call(
+    goto_functionst &goto_functions,
+    const std::string &callercallee,
+    const symbol_tablet &symbol_table,
+    message_handlert &message_handler)
+{
```
Comment (kroening) [inline src/goto-instrument/splice_call.cpp]: We usually use two spaces here.

**Review by kroening — CHANGES_REQUESTED** [review]

**Comment by kroening** [comment]
BTW, how closely related is this to the --function-enter instrumentation?
Could these be consolidated?

**Comment by KPouliasis** [comment]
Hi Daniel, good point. It actually is. I was wondering if this could be made available so that Elizabeth could use my tool (that depends on it) and I will refactor my implementation to use code in function.cpp when I am back to school in a week.

**Review by kroening — APPROVED** [review]

**Pull request merged**